tests/components/rfxtrx/test_sensor.py

```diff
@@ -226,83 +212,93 @@ async def test_update_of_sensors(hass, rfxtrx):
-
-    assert 2 == len(rfxtrx_core.RFX_DEVICES)
-
-    device_num = 0
-    for id in rfxtrx_core.RFX_DEVICES:
-        if id == "sensor_06_01":
-            device_num = device_num + 1
-            assert len(rfxtrx_core.RFX_DEVICES[id]) == 2
-            _entity_temp = rfxtrx_core.RFX_DEVICES[id]["Temperature"]
-            _entity_hum = rfxtrx_core.RFX_DEVICES[id]["Humidity"]
-            assert UNIT_PERCENTAGE == _entity_hum.unit_of_measurement
-            assert 15 == _entity_hum.state
-            assert {
-                "Battery numeric": 9,
-                "Temperature": 51.1,
-                "Humidity": 15,
-                "Humidity status": "normal",
-                "Humidity status numeric": 2,
-                "Rssi numeric": 6,
-            } == _entity_hum.device_state_attributes
-            assert "Bath" == _entity_hum.__str__()
-
-            assert TEMP_CELSIUS == _entity_temp.unit_of_measurement
-            assert 51.1 == _entity_temp.state
-            assert {
-                "Battery numeric": 9,
-                "Temperature": 51.1,
-                "Humidity": 15,
-                "Humidity status": "normal",
-                "Humidity status numeric": 2,
-                "Rssi numeric": 6,
-            } == _entity_temp.device_state_attributes
-            assert "Bath" == _entity_temp.__str__()
-        elif id == "sensor_05_02":
-            device_num = device_num + 1
-            entity = rfxtrx_core.RFX_DEVICES[id]["Temperature"]
-            assert TEMP_CELSIUS == entity.unit_of_measurement
-            assert 13.3 == entity.state
-            assert {
-                "Humidity status": "normal",
-                "Temperature": 13.3,
-                "Rssi numeric": 6,
-                "Humidity": 34,
-                "Battery numeric": 9,
-                "Humidity status numeric": 2,
-            } == entity.device_state_attributes
-            assert "Test" == entity.__str__()
-
-    assert 2 == device_num
-    assert 2 == len(rfxtrx_core.RFX_DEVICES)
+    state = hass.states.get("sensor.test_temperature")
+    assert state
+    assert state.state == "unknown"
+    assert (
+        state.attributes.items()
+        >= {
+            "friendly_name": "Test Temperature",
+            "unit_of_measurement": TEMP_CELSIUS,
+        }.items()
+    )
+
+    state = hass.states.get("sensor.bath_temperature")
+    assert state
+    assert state.state == "unknown"
+    assert (
+        state.attributes.items()
+        >= {
+            "friendly_name": "Bath Temperature",
+            "unit_of_measurement": TEMP_CELSIUS,
+        }.items()
+    )
+
+    state = hass.states.get("sensor.bath_humidity")
+    assert state
+    assert state.state == "unknown"
+    assert (
+        state.attributes.items()
+        >= {
+            "friendly_name": "Bath Humidity",
+            "unit_of_measurement": UNIT_PERCENTAGE,
+        }.items()
+    )
+
+    assert len(hass.states.async_all()) == 3
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5802827)